### PR TITLE
Add getenv and strchr

### PR DESCRIPTION
**Pull request by kant2002**
### After merge
- [ ] If we've decided to leave the current implementation of `getenv`, then open a new issue on the improvements (see the comments for this PR for details)

### CODE REVIEW

**Comment by kant2002** [comment]
This is interesting case for me with `getenv`. It seems that we need to populate some internal structure in our entry point for environment and then operate on that local data. I prefer somehow avoid that for foreseeable future. Maybe you can guess what kind of post processing needed for Lunix/Mac?

**Comment by ForNeVeR** [comment]
Linux tests fail in interesting manner:
```
Expected: ···ocal/games:/snap/bin\n
Actual:   ···ocal/games:/snap/bin�t4�\n
```

Will you take a look at this?

**Comment by ForNeVeR** [comment]
> This is interesting case for me with `getenv`. It seems that we need to populate some internal structure in our entry point for environment and then operate on that local data. I prefer somehow avoid that for foreseeable future. Maybe you can guess what kind of post processing needed for Lunix/Mac?

Ok, I see the issue here. The current implementation allocates a new string on each call, while the C standard implies that we reuse the same memory block for the program environment.

I believe it's okay to leave the current implementation you wrote for now, but open a new issue for its improvement.

The _improvement_ in question, I believe, would be to:
- on the first call to `getenv`, allocate a big memory block for all the environment strings, and remember the indices for each variable somewhere (i.e. in `Dictionary<string, int>`)
- put all the variables into the block and separate with `\0`
- terminate with `\0\0`

This will more or less mimic how glibc works, prevent any leaks, and avoid unnecessary allocations. We may need to reconsider how it all works in the future (in particular, when to invalidate the buffer returned by `getenv()`), if we decide to add functions that modify the environment variables (interestingly, they aren't included in the standard, even though they are in POSIX and alike, so we'll likely want to add them eventually anyway).

Also, it will open the road for alternate POSIX-compatible `main` signature (i.e. the `main(argc, argv, envp)`).

The standard allows us to invalidate the buffer after the next call to `getenv()`, though I think there may be programs that rely on it not being invalidated.

File: Cesium.Runtime/StdLibFunctions.cs [hunk 1]
```diff
@@ -138,6 +138,17 @@ public static int Atoi(byte* ptr)
...
 
+    public static byte* GetEnv(byte* ptr)
+    {
+        var str = StdIoFunctions.Unmarshal(ptr);
+        if (str is null)
+        {
+            return null;
+        }
+
+        return StdIoFunctions.MarshalStr(Environment.GetEnvironmentVariable(str)?.TrimEnd('\\'));
```
Comment (ForNeVeR) [inline Cesium.Runtime/StdLibFunctions.cs:149]: Why `TrimEnd('\\')`? I don't think it's normal for a runtime environment to strip trailing slashes from a variable like this.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

File: Cesium.Runtime/StdLibFunctions.cs (hunk 1 above)
Comment (kant2002) [reply Cesium.Runtime/StdLibFunctions.cs:149]: Similar to https://github.com/ForNeVeR/Cesium/pull/471#issuecomment-1807255956 I cannot understand why there different values in C / .NET, Maybe I should try to use other API? anyway I did not have any reasonable ideas, and put this "bad code" to show you the issues. Will try take a look.

Comment (ForNeVeR) [reply Cesium.Runtime/StdLibFunctions.cs:149]: The API you use looks correct. What's the particular difference you observe? Or do you want me to take a look myself?

Comment (kant2002) [reply Cesium.Runtime/StdLibFunctions.cs:149]: I see one `\` at the end of PATH variable. Also on Linux we see `�t4��`. Maybe that's artifact of `printf`, don't know

File: Cesium.Runtime/StdIoFunctions.cs [hunk 2]
```diff
@@ -279,6 +277,24 @@ public static int FPrintF(void* stream, byte* str, void* varargs)
...
+    internal static byte* MarshalStr(string? str)
+    {
+        Encoding encoding = Encoding.UTF8;
+        if (str is null)
+        {
+            return null;
+        }
+
+        var bytes = encoding.GetBytes(str);
+        var storage = (byte*)StdLibFunctions.Malloc((nuint)bytes.Length);
```
Comment (ForNeVeR) [inline Cesium.Runtime/StdIoFunctions.cs:289]: Gotcha!
```suggestion
        var storage = (byte*)StdLibFunctions.Malloc((nuint)bytes.Length + 1);
```

You forgot to allocate space for the terminating null character.

File: Cesium.Runtime/StdLibFunctions.cs (hunk 1 above)
Comment (ForNeVeR) [reply Cesium.Runtime/StdLibFunctions.cs:149]: Regarding `\`, I believe it is normal: `PATH` variable (especially on Windows) in particular may just end with `\`, this is just its value.

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**